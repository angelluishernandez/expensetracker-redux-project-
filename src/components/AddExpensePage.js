import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../redux/actions/expenses";
const AddExpensePage = props => {
	console.log(props);
	return (
		<div className="AddExpensePage">
			<h1>Add Expense</h1>
			<ExpenseForm
				onSubmit={expense => {
					props.dispatch(addExpense(expense));
					props.history.push("/")
				}}
			/>
		</div>
	);
};

export default connect()(AddExpensePage);
