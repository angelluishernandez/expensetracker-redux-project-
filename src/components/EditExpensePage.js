import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense } from "../redux/actions/expenses";

const EditExpensePage = props => {
	console.log(props.expense);
	return (
		<div className="EditExpensePage">
			<ExpenseForm
				onSubmit={expense => {
					props.dispatch(editExpense(props.expense.id, expense));
					props.history.push("/");
				}}
				expense={props.expense}
			/>
		</div>
	);
};

const mapStateToProps = (state, props) => {
	return {
		expense: state.expenses.find(
			expense => expense.id === props.match.params.id
		),
	};
};

export default connect(mapStateToProps)(EditExpensePage);
