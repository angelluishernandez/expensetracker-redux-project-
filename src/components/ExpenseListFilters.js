import React from "react";
import { connect } from "react-redux";
import {
	setTextFilter,
	sortByAmount,
	sortByDate,
	setStartDate,
	setEndDate,
} from "../redux/actions/filter";

import { DateRangePicker } from "react-dates";

export class ExpenseListFilters extends React.Component {
	state = {
		calendarFocused: null,
	};

	onDatesChange = ({ startDate, endDate }) => {
		this.props.setStartDate(startDate);
		this.props.setEndDate(endDate);
	};

	onFocusChange = calendarFocused => {
		this.setState({
			calendarFocused,
		});
	};

	onTextChange = e => {
		this.props.setTextFilter(e.target.value);
	};

	onSortChange = e => {
		if (e.target.value === "date") {
			this.props.sortByDate();
		} else if (e.target.value === "amount") {
			this.props.sortByAmount();
		}
	};
	render() {
		return (
			<div className="ExpanseListFilters">
				<input type="text" defaultValue={this.props.filters.text} />
				<select value={this.props.filters.sortBy}>
					<option value="date">Date</option>
					<option value="amount">Amount</option>
				</select>
				<DateRangePicker
					startDate={this.props.filters.startDate}
					endDate={this.props.filters.endDate}
					onDatesChange={this.onDatesChange}
					focusedInput={this.state.calendarFocused}
					onFocusChange={this.onFocusChange}
					numberOfMonths={1}
					isOutsideRange={() => false}
					showClearDates={true}
					startDateId="startDateId"
					endDateId="endDateId"
				/>
			</div>
		);
	}
}

const mapStateToProps = state => {
	return {
		filters: state.filters,
	};
};

const mapDispatchToProps = dispatch => ({
	setTextFilter: text => dispatch(setTextFilter(text)),
	sortByDate: () => dispatch(sortByDate()),
	sortByAmount: () => dispatch(sortByAmount()),
	setStartDate: startDate => dispatch(setStartDate(startDate)),
	setEndDate: endDate => dispatch(setEndDate(endDate)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
