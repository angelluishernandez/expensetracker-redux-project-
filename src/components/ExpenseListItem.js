import React from "react";
import { connect } from "react-redux";
import { removeExpense } from "../redux/actions/expenses";
import { Link } from "react-router-dom";

export const ExpenseListItem = ({ description, amount, createdAt, id, dispatch }) => {
	return (
		<div className="ExpenseListItem">
			<Link to={`/edit/${id}`}>
				<p>{description}</p>
			</Link>
			<p>
				${amount} - {createdAt}
			</p>
			<button
				onClick={() => {
					dispatch(removeExpense({ id }));
				}}
			>
				Remove
			</button>
		</div>
	);
};

const mapStateToProps = state => {
	return {
		expenses: state.expenses,
	};
};

export default connect(mapStateToProps)(ExpenseListItem);
