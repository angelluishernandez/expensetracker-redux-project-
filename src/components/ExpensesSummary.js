import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpenses from "../redux/selectors/expenses.selector";
import selectExpensesTotal from "../redux/selectors/expenses-total.selector";

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
	const expenseWord = expensesCount === 1 ? "expense" : "expenses";
	const formatedExpensesTotal = numeral(expensesTotal).format("$0,0.00");
	return (
		<div className="ExpensesSummary ">
			<h1>
				Viewing {expensesCount} {expenseWord} totalling {formatedExpensesTotal}
			</h1>
		</div>
	);
};

const mapStateToProps = state => {
	const visibleExpenses = selectExpenses(state.expenses, state.filters);
	return {
		expensesCount: visibleExpenses.length,
		expensesTotal: selectExpensesTotal(visibleExpenses)
	};
};

export default connect(mapStateToProps)(ExpensesSummary);
