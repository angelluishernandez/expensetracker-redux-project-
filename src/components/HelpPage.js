import React from "react";

const HelpPage = (props) => <div className="HelpPage">HelpPage</div>;

export default HelpPage;
