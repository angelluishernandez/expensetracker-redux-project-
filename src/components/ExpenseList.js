import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from "../redux/selectors/expenses.selector";

export const ExpenseList = props => (
	<div className="ExpenseList">

		{
			props.expenses.length === 0 ? (props.expenses.map((expense, index) => {
				return <ExpenseListItem key={index} {...expense} />;
			})) : (<p>No expenses</p>)



		}
		

	</div>
);

// Best way to do it

const mapStateToProps = state => {
	return {
		expenses: selectExpenses(state.expenses, state.filters),
	};
};

export default connect(mapStateToProps)(ExpenseList);

// Simple way to do it

// const ConnectedExpenseList = connect((state)=> {
//   return {
//     expenses: state.expenses,

//   }
// })(ExpenseList )
// export default ConnectedExpenseList;
