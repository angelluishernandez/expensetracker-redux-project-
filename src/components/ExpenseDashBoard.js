import React from "react";
import ExpenseList from "./ExpenseList";
import ExpanseListFilters from "./ExpenseListFilters";

const ExpenseDashBoard = props => {
	return (
		<div className="ExpenseDashBoard">
			<h1>Expense Dashboard</h1>
			<ExpanseListFilters />
			<ExpenseList />
		</div>
	);
};
export default ExpenseDashBoard;
