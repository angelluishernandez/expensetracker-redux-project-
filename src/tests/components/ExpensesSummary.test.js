import React from "react";
import { configure } from "enzyme";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import { ExpensesSummary } from "../../components/ExpensesSummary";

configure({ adapter: new Adapter() });

test("should correctly render expenses summary with one expense", () => {
	const wrapper = shallow(
		<ExpensesSummary expensesCount={1} expensesTotal={235} />
	);
	expect(toJSON(wrapper)).toMatchSnapshot();
});

test("should correctly render expenses summary with multiple expenses", () => {
	const wrapper = shallow(
		<ExpensesSummary expensesCount={23} expensesTotal={28643846384635} />
	);
	expect(toJSON(wrapper)).toMatchSnapshot();
});
