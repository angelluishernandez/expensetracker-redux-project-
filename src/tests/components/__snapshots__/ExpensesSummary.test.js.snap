// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $286,438,463,846.35
  </h1>
</div>
`;

exports[`should correctly render expenses summary with one expense 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $2.35
  </h1>
</div>
`;
